### PR TITLE
MM-45520: Run Details Page - RHS Info links

### PR DESCRIPTION
**Pull request by agarciamontoro**
#### Summary

This PR adds some links from the RHS:
- Key Metrics links now to the Retrospective in the center section
  - Each metric in the Key Metrics links now to the specific metric in the center section, automatically focusing it. 
- Recent Activity opens the Timeline in the RHS, with a back button to get to the RHS Info.

The video shows all new interactions:

https://user-images.githubusercontent.com/3924815/177977355-f220a5c6-7059-4163-b3e5-193f22bf671a.mp4

As you see, I added a new dependency, `react-router-hash-link`. This is to overcome a slight problem that would have needed a more complex solution: if we use plain links with hashes on them, clicking on `View Retrospective` would work the first time, because it changes the URL, but if the user scrolls again to other part of the page and click one more time on `View Retrospective`, that second click does nothing, as the URL did not change. `react-router-hash-link` takes care of all those edge cases for us, and it just works.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-45520

#### Checklist

- [ ] ~~Telemetry updated~~
- [ ] ~~Gated by experimental feature flag~~
- [ ] ~~Unit tests updated~~

### CODE REVIEW

**Review by trilopin — APPROVED** [review]
Hey, the code and feature are great.

I left 2 optional comments

**Review by iomodo — APPROVED** [review]
LGTM! Thanks @agarciamontoro

File: webapp/src/components/backstage/playbook_runs/playbook_run/rhs_info_styles.tsx [hunk 1]
```diff
@@ -15,21 +15,36 @@ export const Section = styled.section`
     padding: 24px 0;
 `;
 
+type LinkType = {to: string, name: string} | {onClick: () => void, name: string};
```
[the remark this answers is not in the document]
Comment (agarciamontoro) [reply webapp/src/components/backstage/playbook_runs/playbook_run/rhs_info_styles.tsx]: Done in https://github.com/mattermost/mattermost-plugin-playbooks/pull/1302/commits/ef1125d0804c0589a2b7d6380b0e2f7eff99c190. I ended up using type guards, since `link instanceof LinkURL` cannot be used, because LinkURL is not a value.

File: webapp/src/components/backstage/playbook_runs/playbook_run_backstage/metrics/metric_input.tsx [hunk 2]
```diff
@@ -31,12 +35,22 @@ const MetricInput = ({
...
-    inputRef: textareaRef,
     onChange,
     disabled,
+    autofocus,
+    onClickOutside,
 }: Props) => {
     const {formatMessage} = useIntl();
 
+    const textareaRef = useRef<HTMLInputElement | null>(null);
+    useClickOutsideRef(textareaRef, onClickOutside ?? (() => {/*do nothing*/}));
```
[the remark this answers is not in the document]
Comment (agarciamontoro) [reply webapp/src/components/backstage/playbook_runs/playbook_run_backstage/metrics/metric_input.tsx]: You're completely right. Done in https://github.com/mattermost/mattermost-plugin-playbooks/pull/1302/commits/966257d9139344d99ba1c89103c25dfa0e7e316b.
